_plugins/versions_generator.rb

```diff
@@ -0,0 +1,41 @@
+require 'json'
+require 'yaml'
+
+module Jekyll
+  class VersionsGenerator < Generator
```

Comment:
Custom plugins can't be run on GitHub Pages, I don't think it's a good idea to add one.